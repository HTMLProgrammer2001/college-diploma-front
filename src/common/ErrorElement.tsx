import React from 'react';
import i18next from 'i18next';


type IErrorProps = {
	error?: string
};

const ErrorElement: React.FC<IErrorProps> = ({error}) => (
	<div>{error || i18next.t('common.errorOccurred')}</div>
);

export default ErrorElement;
