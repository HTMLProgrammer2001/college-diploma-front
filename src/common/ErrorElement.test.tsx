import React from 'react';
import {shallow} from 'enzyme';

import ErrorElement from './ErrorElement';
import i18next from 'i18next';


describe('Test error element', () => {
	it('Test render', () => {
		const wrapper = shallow(<ErrorElement/>);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Test with error', () => {
		const error = 'ERROR';
		const wrapper = shallow(<ErrorElement error={error}/>);

		expect(wrapper.text()).toBe(error);
	});

	it('Test without error', () => {
		const wrapper = shallow(<ErrorElement/>);
		expect(wrapper.text()).toBe(i18next.t('common.errorOccurred'));
	});
});
