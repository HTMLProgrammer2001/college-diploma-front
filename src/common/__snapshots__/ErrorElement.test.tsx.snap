// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test error element Test render 1`] = `"<div>common.errorOccurred</div>"`;
